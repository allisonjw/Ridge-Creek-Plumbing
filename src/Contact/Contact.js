import React from 'react';
import './Contact.scss';
import contactImage from '../images/boots_gloves.jpeg';
import { FaFacebookSquare, FaLinkedin, FaTwitterSquare, FaInstagram } from "react-icons/fa";
import Fade from 'react-reveal/Fade';

export const Contact = () => {
    return (
      <section className="section_contact">
        <Fade left>
        <div className="contact_banner" 
        aria-label="notification of contact page">
          <div className="contact_img" aria-label="picture of hard hat and boots" style={{ backgroundImage:`linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(${contactImage})` }}>
            <h2 className="contact_banner-h2">
              Contact Us
            </h2>
          </div>
        </div>
        </Fade>
        <section className="contact_body">
          <div className="location_div" aria-label="container for location information">
            <h3 className="contact_h3">LOCATION</h3>
            <p className="contact_address-paragraph">
              1234 Main Street
              <br></br>
              Olympia, WA 98502
            </p>
          </div>
          <div className="email_div" aria-label="container for contact email">
            <h3 className="contact_h3">EMAIL</h3>
            <a    
              href="mailto:Ridgecreekplumbingllc@gmail.com" 
              target="_blank" 
              className="contact_link"
              aria-label="email address"
              rel="noopener noreferrer">
              Ridgecreekplumbingllc@gmail.com
            </a>
          </div>
          <div className="phone_div" aria-label="container for contact telephone number">
            <h3 className="contact_h3">PHONE</h3>
            <a 
              href="tel:3604856605" 
              className="contact_link"
              aria-label="phone number">
              360.485.6605
            </a>
          </div>
        </section>
        <section className="contact_socialMedia">
          <h3 className="social_h3">Contact Us On Social Media</h3>
          <div className="contact_media-holder" aria-label="container for all social media links">
            <a href="https://www.facebook.com/ridgecreekplumbing/"
              className="contact_socialMedia-img" 
              target="_blank" 
              rel="noopener noreferrer">
              <FaFacebookSquare />
            </a>
            <a href="instagram" 
              className="contact_socialMedia-img" 
              target="_blank" 
              rel="noopener noreferrer">
              <FaInstagram />
            </a>
            <a href="linkedin"
              className="contact_socialMedia-img"  
              target="_blank" 
              rel="noopener noreferrer">
              <FaLinkedin />
            </a> 
            <a href="twitter"
              className="contact_socialMedia-img"  
              target="_blank" 
              rel="noopener noreferrer">
              <FaTwitterSquare />
            </a>
          </div>   
        </section>
      </section>
    )
};

export default Contact;