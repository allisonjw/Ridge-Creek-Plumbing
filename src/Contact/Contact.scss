@import '../SCSS/variables';
@import '../SCSS/media-queries';

.section_contact {
    animation: fade-in 1s ease-out both 0.3s;
    -webkit-animation: fade-in 1s ease-out both 0.3s;
    -moz-animation: fade-in 1s ease-out both 0.3s;
    -o-animation: fade-in 1s ease-out both 0.3s;
}

.contact_banner {
    width: 90%;
    height: 70vh;
    margin: 2em auto;
    position: relative;
    border-bottom: 1px solid #000;
    padding-bottom: 1%;
}

.contact_img {
    width: 100%;
    height: 70vh;
    background-size: cover;
    background-attachment: fixed;
    background-position: 50% 50%;
    background-repeat: no-repeat;
    .contact_banner-h2 {
        position: absolute;
        font-size: 4em;
        color: white;
        top: 18%;
        left: 14%;
        text-shadow: 5px 4px 6px #000000b5;
    }
}

.contact_body {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    flex-wrap: wrap;
    margin-bottom: 7%;
}

.location_div,
.email_div,
.phone_div {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    padding: 1% 3% 1% 0;
    align-items: center;
}

.location_div,
.email_div {
    border-right: 1px solid $main_color;
}

.contact_h3 {
    color: $accent_color;
    font-size: 1.7em;
    margin-bottom: 12px;
    letter-spacing: 2px;;
}

.contact_link {
    text-decoration: none;
    transition: all .2s ease;
    font-size: 1.4em;
    color: $main_color;
    font-weight: 500;
    &:hover {
        cursor: pointer;
        transform: scale(1.1);
    }
}

.contact_address-paragraph {
    font-size: 1.4em;
    line-height: 1.4;
    text-align: center;
    font-weight: 500;
    color: $main_color;
}

.contact_socialMedia {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    .social_h3 {
        font-size: 2em;
        margin-bottom: 1.5%;
    }
    .contact_socialMedia-img {
        color: $accent_color;
        font-size: 3em;
        margin: 0 1%;
          &:hover {
          cursor: pointer;
        }
    }
}

.contact_media-holder {
    display: flex;
    flex-direction: row;
}

@keyframes fadeIn {
    from {opacity: 0%;}
    to {opacity: 100%;}
}

