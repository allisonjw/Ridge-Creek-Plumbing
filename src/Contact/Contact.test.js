import React from 'react';
import Contact from './Contact';
import { shallow } from 'enzyme';

describe('Contact', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Contact />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});