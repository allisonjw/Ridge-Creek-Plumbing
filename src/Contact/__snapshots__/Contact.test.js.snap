// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match the snapshot 1`] = `
<section
  className="section_contact"
>
  <Fade
    left={true}
  >
    <div
      aria-label="notification of contact page"
      className="contact_banner"
    >
      <div
        aria-label="picture of hard hat and boots"
        className="contact_img"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(255, 252, 252, 0.25), rgba(255, 254, 254, 0.15)), url(boots_gloves.jpeg)",
          }
        }
      >
        <h2
          className="contact_banner-h2"
        >
          Contact Us
        </h2>
      </div>
    </div>
  </Fade>
  <section
    className="contact_body"
  >
    <div
      aria-label="container for location information"
      className="location_div"
    >
      <h3
        className="contact_h3"
      >
        LOCATION
      </h3>
      <p
        className="contact_address-paragraph"
      >
        2849 Black Lake Blvd.
        <br />
        Olympia, WA 98502
      </p>
    </div>
    <div
      aria-label="container for contact email"
      className="email_div"
    >
      <h3
        className="contact_h3"
      >
        EMAIL
      </h3>
      <a
        aria-label="email address"
        className="contact_link"
        href="mailto:Ridgecreekplumbingllc@gmail.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Ridgecreekplumbingllc@gmail.com
      </a>
    </div>
    <div
      aria-label="container for contact telephone number"
      className="phone_div"
    >
      <h3
        className="contact_h3"
      >
        PHONE
      </h3>
      <a
        aria-label="phone number"
        className="contact_link"
        href="tel:3604856605"
      >
        360.485.6605
      </a>
    </div>
  </section>
  <section
    className="contact_socialMedia"
  >
    <h3
      className="social_h3"
    >
      Contact Us On Social Media
    </h3>
    <div
      aria-label="container for all social media links"
      className="contact_media-holder"
    >
      <a
        className="contact_socialMedia-img"
        href="https://www.facebook.com/ridgecreekplumbing/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FaFacebookSquare />
      </a>
      <a
        className="contact_socialMedia-img"
        href="instagram"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FaInstagram />
      </a>
      <a
        className="contact_socialMedia-img"
        href="linkedin"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FaLinkedin />
      </a>
      <a
        className="contact_socialMedia-img"
        href="twitter"
        rel="noopener noreferrer"
        target="_blank"
      >
        <FaTwitterSquare />
      </a>
    </div>
  </section>
</section>
`;
